src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,56 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+/**
+ * Edits the particulars of a person whose name matches exactly as typed in the command.
+ * Keyword matching is case sensitive.
+ *
+ * NOTE: This class function only returns "found" if the name is valid, or "not found", if otherwise.
+ *       For purpose of CS2113 LO Week 5.11
+ */
+public class EditCommand extends Command{
+    public static final String COMMAND_WORD = "edit";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edit a person's profile according to his/her ID.\n" +
+            "Parameters: [ID] [PARAMETER TO EDIT] [NEW INPUT]\n" +
+            "Example: " + COMMAND_WORD + " 3 e/ johndoe@gmail.com";
+
+    private String searchName;
+
+
+    /**
+     * Constructor for EditCommand
+     * @param targetVisibleIndex
+     */
+    public EditCommand(int targetVisibleIndex) {
+        super(targetVisibleIndex);
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            if (!addressBook.containsPerson(target)) {
+                return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
+            }
+            return new CommandResult("This person is found.");
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+    }
+
+    /**
+     * Checks if the specified person name is inside the address book.
+     * @param searchName to check if the name is inside the address book
+     * @return true if the person is found, and false if otherwise.
+     */
+    protected boolean isPersonFound(String searchName) {
```

Comment:
Good header documentation/comment for your function as well, to indicate what the function will do, the parameters it takes in and the results it will return.